packages/core/upload/admin/src/pages/App/MediaLibrary.js

```diff
@@ -105,19 +108,32 @@ export const MediaLibrary = () => {
 
   const assets = assetsData?.results;
   const assetCount = assets?.length ?? 0;
-
   const isNestedFolder = !!query?.folder;
+
   const isLoading =
     foldersLoading || folderStructureIsLoading || permissionsLoading || assetsLoading;
+  const isFolder = !isLoading && query.folder;
+
+  const parentFolderId = isFolder && findRecursiveParentFolderId(folderStructure[0], query.folder);
+  const backQuery = parentFolderId
+    ? stringify({ ...query, folder: parentFolderId }, { encode: false })
+    : stringify(
+        { sort: query.sort, page: query.page, pageSize: query.pageSize },
+        { encode: false }
+      );
```

Comment:
Why do we need a default case, in case `parentFolderId` is not defined? On the root level we don't have to display the back button.

Reply:
this default case is for the folders nested 1 level inside the ML, the back button should send you back to the root of the ML! do you have a better idea to handle that?